spec/factories/project_steps.rb

```diff
@@ -0,0 +1,49 @@
+FactoryGirl.define do
+  factory :project_step do
+    # project_table 'Loan'
+    # project_id { create(:loan).id }
+    project { create(:loan) }
+    person
+    # need to make sure parent saved before assigning these
+    # summary { Faker::Lorem.sentence }
+    # details { Faker::Lorem.paragraph }
+    scheduled_date { Faker::Date.between(Date.civil(2004, 01, 01), Date.today) }
+    completed_date { Faker::Date.between(Date.civil(2004, 01, 01), Date.today) }
+    is_finalized [true,false].sample  ##JE todo: is ther a better pattern to assign a boolean?
+    type_option_id { ProjectStep::TYPE_OPTIONS.values.sample }
+
+    trait :completed do
+      completed_date { Faker::Date.between(date, Date.today) }
+    end
+
+    trait :past do
+      scheduled_date { Faker::Date.backward }
+    end
+
+    trait :future do
+      scheduled_date { Faker::Date.forward }
+    end
+
+    # is this even valid?
+    # trait :for_loan do
+    #   transient { loan_id 0 }
+    #   after(:build) do |event, evaluator|
```

Comment:
Yeah, the production db was also full of references to 0 id loans.  I removed the commented code because I don't believe this is valid moving forward.  Also, ProjectSteps are inherently 'for_loan' now, so no need for a distinct trait.